### ISSUE TITLE
aggregate-columns might lose the order of columns

### ISSUE THREAD

**Issue by daslu** [scicloj/tablecloth#35]
Tested with `scicloj/tablecloth {:mvn/version "5.05"}`:

```clj

(-> (array-map
     :a [1 1]
     :b 2
     :c 3
     :d 4
     :e 5
     :f 6
     :g 7
     :h 8
     :i 9)
    tablecloth.api/dataset
    (tablecloth.api/aggregate-columns [:a :b :c :d :e :f :g :h :i]
                                      tech.v3.datatype.functional/sum))

_unnamed [1 9]:

|       :e |       :g |       :c |       :h |       :b |       :d |       :f |       :i |       :a |
|----------|----------|----------|----------|----------|----------|----------|----------|----------|
|     10.0 |     14.0 |      6.0 |     16.0 |      4.0 |      8.0 |     12.0 |     18.0 |      2.0 |

```

**Comment by genmeblog** [comment]
Probably there is a regular map along the path.

**Comment by genmeblog** [comment]
@daslu the issue lays on the TMD side. `mapseq-reader` uses regular HashMap to store grouping key. It's here: https://github.com/techascent/tech.ml.dataset/blob/master/src/tech/v3/dataset/readers.clj#L46

**Comment by genmeblog** [comment]
(deleted previous comment, which is probably not true in this case... - but may be in other cases)

**Comment by genmeblog** [comment]
Ok, got it working with array-map

```clojure
(-> (array-map
     :a [1 1]
     :b 2
     :c 3
     :d 4
     :e 5
     :f 6
     :g 7
     :h 8
     :i 9)
    dataset
    (aggregate-columns [:a :b :c :d :e :f :g :h :i]
                       tech.v3.datatype.functional/sum))
;; => _unnamed [1 9]:
;;    |       :a |       :b |       :c |       :d |       :e |       :f |       :g |       :h |       :i |
;;    |----------|----------|----------|----------|----------|----------|----------|----------|----------|
;;    |      2.0 |      4.0 |      6.0 |      8.0 |     10.0 |     12.0 |     14.0 |     16.0 |     18.0 |
```

I wonder where we can find another place around this issue.

**Comment by genmeblog** [comment]
fixed in 5.11

**Comment by daslu** [comment]
Wonderful!
